common-api/src/test/java/org/pmiops/workbench/db/model/CommonStorageEnumsTest.java

```diff
@@ -51,4 +55,52 @@ public void testBijectiveStorageMapping() {
       assertThat(v).isEqualTo(fromStorage.apply(storageValue));
     }
   }
+
+  // domain ID is stringly-typed so special-case this
+
+  @Test
+  public void testDomainIdBijectiveStorageMapping() {
+    for (Domain v : Domain.values()) {
+      String storageValue = CommonStorageEnums.domainToDomainId(v);
+      assertWithMessage("unmapped enum value: " + v).that(storageValue).isNotNull();
+      assertThat(v).isEqualTo(CommonStorageEnums.domainIdToDomain(storageValue));
+    }
+  }
+
+  // copied from api/StorageEnumsTest because the above tests are not comprehensive
+  @Test
+  public void noMissingMapEntries() throws Exception {
+    for (Field f : CommonStorageEnums.class.getDeclaredFields()) {
+      if (f.getType() != BiMap.class) {
+        continue;
+      }
+
+      Class enumClass =
+          (Class) ((ParameterizedType) f.getAnnotatedType().getType()).getActualTypeArguments()[0];
+
+      Method enumToShort = null;
+      Method shortToEnum = null;
+      for (Method m : CommonStorageEnums.class.getDeclaredMethods()) {
+        if (m.getParameterTypes()[0].equals(enumClass)) {
+          enumToShort = m;
+        }
+
+        if (m.getReturnType().equals(enumClass)) {
+          shortToEnum = m;
+        }
+      }
+
+      // stringly typed map - test with testDomainIdBijectiveStorageMapping instead
+      if (enumToShort.getName().equals("domainIdToDomain")) {
```

Comment:
Is there a more elegant way?  I want to be sure to exclude only this one, for future-proofing

Reply:
Not worth spending much time on.

Reply:
This looks like a fun exercise, but it's way too abstract/reflexive for test code (or application code). Plus, I want to kill dbStorageEnums soon in favor of Lebowski Enums.

Can we shelve this for another day? I'm happy to pair on the enum stuff. It's a bit funny looking at first, but I think you'll like it.

Reply:
Some more English comments on what each chunk is doing would help. Most of us don't use these kinds of functions every day.

Reply:
I'll do that but in a follow up along with some other issues I've found

Reply:
Comments and fix are at #3159